brion/v2/morphology.cpp

```diff
@@ -0,0 +1,244 @@
+
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "morphology.h"
+
+#include <brion/morphology.h>
+#include <brion/circuit.h>
+
+#include <math.h>
+
+namespace brion
+{
+namespace v2
+{
+
+namespace
+{
+    void nullDeleter( const brion::Morphology* ) {}
+}
+
+class Morphology::_Impl
+{
+public:
+    boost::shared_ptr< const brion::Morphology > _morphology;
+
+    Vector4fsPtr _points;
+    Vector2isPtr _sections;
+    SectionTypesPtr _types;
+    Vector2isPtr _apicals;
+
+    bool _isGlobal;
+    Vector3f _somaPosition;
+
+    _Impl( const boost::shared_ptr< const brion::Morphology >& morphology,
+           const MorphologyRepairStage stage )
+        : _morphology( morphology )
+        , _points( morphology->readPoints( stage ))
```

Comment:
These members are also public. I kept the _ to make it clear they are members. But if you insist I can remove the _.